trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/rest/ParsedS3Request.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.s3.proxy.server.rest;
+
+import com.google.common.base.Splitter;
+import jakarta.ws.rs.core.MultivaluedMap;
+import jakarta.ws.rs.core.UriBuilder;
+import org.glassfish.jersey.server.ContainerRequest;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+
+import static io.trino.s3.proxy.server.collections.MultiMapHelper.lowercase;
+import static java.util.Objects.requireNonNull;
+
+public record ParsedS3Request(String bucketName, String pathInBucket, MultivaluedMap<String, String> requestHeaders, MultivaluedMap<String, String> requestQueryParameters,
```

Comment:
some NITs:

we can call this just `S3Request`, i don't think anything else is using that name. WDYT?

can we have each field on a new line?

Reply:
Refer to https://github.com/trinodb/s3-proxy/pull/51#discussion_r1625885152 for the name, I originally named this `S3RequestInformation`, then we discussed `S3Request` but settled on this.

I don't really have a strong preference between this or `S3Request` so happy for you & @Randgalt to discuss

Reply:
I also don't have a strong preference, so let's go with this for now!